transformer_engine/pytorch/csrc/userbuffers/userbuffers-host.cpp

```diff
@@ -279,8 +282,15 @@ int create_communicator_grouped2(communicator **comm, int pipegpus, int pipenode
                            (NVTE_MAX_SMS + 100) * sizeof(int)));
   for (int i = 0; i < 100 + NVTE_MAX_SMS; i++)
     (*comm)->hostflags[i] = 0;
+#if (defined(__arm__) && __ARM_ARCH >= 7) || defined(__aarch64__)
+  __asm__ __volatile__("dmb sy" : : : "memory");
+#elif defined(_MSC_VER) && _MSC_VER >= 1310
+extern "C" void _ReadWriteBarrier();
+#  pragma intrinsic(_ReadWriteBarrier)
+#else
   _mm_mfence();
   sleep(1);
+#endif
```

Comment:
Thanks a lot for the PR @aurianer 
Yet I think we still need the sleep as I get a segfault without it:
```
0: ==== backtrace (tid:    102) ====
...
0:  3 0x000000000007f3f4 ubuf::UbufP2PCommOverlap::UbufP2PCommOverlap()  /tmp/pip-req-build-l4h7exuy/transformer_engine/pytorch/csrc/comm_gemm_overlap.h:565
```

```suggestion
#endif
  sleep(1);
```